test/common/testlib.py

```diff
@@ -899,6 +941,19 @@ def img_eq(ref, now, delta):
                 print("Differences in pixel test " + base)
                 self.failed_pixel_tests += 1
 
+    def assert_pixels(self, selector, key, ignore=[], skip_layouts=[]):
+        """Compare the given element with its reference in all layouts"""
+
+        if not (Image and self.pixels_label and self.cdp and self.cdp.valid):
```

Comment:
I know it's not new code, we already have the same condition in assert_pixels_in_current_layout() -- but why the `cdp` and `cdp.valid`? That just seems to hide broken tests, where we try to do an assert_pixels() before initializing/logging inot the browser. This could lead to pixel tests which never run, silently ignoring them. Let's please error out in that case.

Reply:
Good point.  This comes originally from `Browser.snapshot`, which needs to be much more robust and not blow up when a test fails without a working browser.  `assert_pixel` can assume a working browser.